SamX WIP PR

SamAcctX opened the pull request:
Opened for easy diff view

SamAcctX committed:
Initial pass at docker build for bot part - TBD on DB back-end

SamAcctX committed:
Temp disabling the github issues bot stuff - don't feel like fucking with API keys for that yet...

SamAcctX committed:
Updated docker to download game objects from Google Drive and extract at startup

SamAcctX committed:
Setting gdown to quiet to help keep logs cleaner

Trimatix commented on Dockerfile:
```diff
@@ -0,0 +1,122 @@
+
+# Might be able to change this to something a bit smaller (like runtime instead of devel) 
+# from https://hub.docker.com/r/nvidia/cuda/tags
+FROM nvidia/cuda:${CUDA_VERSION}-cudnn-devel-ubuntu${UBUNTU_RELEASE} AS base
```
my VPS doesn't have a GPU, would that be an issue?

Trimatix replied:
actually looking at this, the nvidia stuff might be from your AI work

SamAcctX replied:
Yeah, it will be fine without a GPU.  This is just the base image I use for a lot of other stuff, so was easiest to use it as a starting point.  Just take out the "Deploy" part in the docker-compose.yml since you don't have a GPU there.

Trimatix commented on bot/cfg/cfg.py:
```diff
@@ -203,7 +203,8 @@ def cogPath(cogName: str, basePackage: str = "bot.cogs") -> str:
     cogPath("EmbedEditorCog", basePackage="bot.cogs.util"),
-    cogPath("GithubUtilCog", basePackage="bot.cogs.util"),
+    # Temp disable the github issue stuff for now...
+    # cogPath("GithubUtilCog", basePackage="bot.cogs.util"),
```
just in case it's easier for you - we can also disable cogs from the toml config file, like the one in the config repo

Trimatix commented on docker-compose.yml:
```diff
@@ -0,0 +1,75 @@
+      - /etc/localtime:/etc/localtime:ro
+      - /etc/timezone:/etc/timezone:ro
+      # Keep data across rebuilds...
+      - ./mappings/bountybot/saveData:/app/bountybot/saveData
```
In case this changes anything: in the not-too-distant future, we'll have a 'workshop' similar to the steam workshop, where players can have custom items added to the game. The bot will add assets for those items to the config files directory.

SamAcctX replied:
Should be doable can either use a mapped folder like this or define a docker volume and mount the volume to the container.  The latter is the "proper" way to do this, but I like mapped folders as I can see/work with the contents more easily.

Trimatix commented:
Awesome work, this will make hosting so much easier. Where we currently download assets from google drive, it might make sense to provide a github PAT, and have the setup script clone the config files repo instead

SamAcctX commented:
> Awesome work, this will make hosting so much easier. Where we currently download assets from google drive, it might make sense to provide a github PAT, and have the setup script clone the config files repo instead

Thanks!  Eventually, I want to get away from config files and rely almost exclusively on environment variables (at least for stuff that is likely to change).  And now that the massive stuff is downloaded at container start, we should be able to make a "sample" toml and have it all on one repo

SamAcctX committed:
clean version of config files - need to update readme tho...

SamAcctX committed:
Revert "Temp disabling the github issues bot stuff - don't feel like fucking with API keys for that yet..."

This reverts commit c297cdaf41063783775c23f5870cf9b38c4167ad.

SamAcctX committed:
Separate game object meta and classes from static content (like skin/blender assets)

SamAcctX committed:
Some updated for github

SamAcctX committed:
Fix blender imports for blender 4.x

SamAcctX committed:
Fix /stats, and hopefully the duel request as well (to be verified)

SamAcctX committed:
Seems like latest version of emojis works - at least so far...

SamAcctX committed:
mappings shouldn't be in the base image - oops...

SamAcctX committed:
Starting on refactoring imports to use full names instead of relative pathing...

SamAcctX committed:
Should cover all imports in the bot.baseClasses folder (and any files with matching import patterns)

SamAcctX committed:
upping max render res to 4k

SamAcctX committed:
Should cover all imports in the bot.cfg folder (and any files with matching import patterns)

SamAcctX committed:
Some progress on rendering fixes... I think...

SamAcctX committed:
Some progress on rendering fixes... I think...

SamAcctX committed:
More updates to full-path imports - hope nothing breaks...

SamAcctX committed:
fix one import oops - bound to happen sooner or later

SamAcctX committed:
Add CUDA GPU offloading for blender - first render takes longer tho to compile the kernel

SamAcctX committed:
Updating startup script to pre-compile cuda kernel for blender

SamAcctX committed:
Another round of import updates

SamAcctX committed:
Should be the last of import refactoring, time to smoke test...

SamAcctX committed:
re-enabling blender CUDA compile in startup script

SamAcctX committed:
GPU and non-GPU docker compose files - non-GPU will default

SamAcctX commented:
Ok, I think my work on this branch (or at least this iteration) is done.  As far as I can tell, all the original functionality still works (and a few items were fixed).

- Added containerization (docker)
- Updated Blender to 4.x
- Added GPU offloading for blender (if CUDA device is present)
- Docker startup script will run a "warmup" blender render if a GPU is detected to pre-compile the CUDA kernels for blender so that later renders will start instantly
  - In testing, first render of a 4k image took 4 mins to compile the kernel, then 12s to do the render.  Subsequent renders only took the 12s as the kernel was already compiled)
- Upped max resolution for skins to be 4k (tho the default of roughly 320p remains)
- Refactored all imports to use complete import paths instead of relative paths
- Disabled Github integration for issue management (until it's actually worth re-enabling and trying to get a new API key)
- Added all emoji assets and a PoSh script that can upload them as application emojis (no reliance on having them on some random discord server somewhere)
- Added the rendering meta files to source control (will eventually be removed once DB support is fully done)
- Docker startup script to download and overlay ship skin assets on container startup (if not present - will not re-download on a simple container restart)
